src/ape/api/providers.py

```diff
@@ -60,17 +60,40 @@ class BlockAPI(BaseInterfaceModel):
     # NOTE: All fields in this class (and it's subclasses) should not be `Optional`
     #       except the edge cases noted below
 
+    """
+    The number of transactions in the block.
+    """
     num_transactions: int = 0
+
+    """
+    The block hash identifier.
+    """
     hash: Optional[Any] = None  # NOTE: pending block does not have a hash
+
+    """
+    The block number identifier.
+    """
     number: Optional[int] = None  # NOTE: pending block does not have a number
+
+    """
+    The preceeding block's hash.
+    """
     parent_hash: Any = Field(
         EMPTY_BYTES32, alias="parentHash"
     )  # NOTE: genesis block has no parent hash
-    size: int
+
+    """
+    The timestamp the block was produced.
+    """
     timestamp: int
```

Comment:
Believe this note makes sense (double check me)

```suggestion
    timestamp: int  #  NOTE: pending block uses current timestamp
```

Reply:
this might make more sense in the doc str so users can see it from the doc-site as well

Reply:
another thing: this is at the API level- so technically speaking, a Provider that implements ProviderAPI could handle this differently, especially on different ecosystems.